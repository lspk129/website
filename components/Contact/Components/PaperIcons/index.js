import Paper from 'material-ui/Paper';
import styled from 'styled-components';

const PaperIcon = styled(Paper)`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 50px;
  height: 50px;
`;

export default PaperIcon;
