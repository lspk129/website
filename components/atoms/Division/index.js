import styled from 'styled-components';

const Division = styled.div`
  ${props => props.textAlign && `text-align: ${props.textAlign}`};
  // margin properties
  ${props => props.m && `margin: ${props.m}rem;`}
  ${props => props.mx && `margin: 0rem ${props.mx}rem;`}
  ${props => props.my && `margin: ${props.my}rem 0rem;`}
  ${props => props.mt && `margin-top: ${props.mt}rem;`}
  ${props => props.mb && `margin-bottom: ${props.mb}rem;`}
  ${props => props.mr && `margin-right: ${props.mr}rem;`}
  ${props => props.ml && `margin-left: ${props.ml}rem;`}
  // padding properties
  ${props => props.p && `padding: ${props.p}rem;`}
  ${props => props.px && `padding: 0rem ${props.px}rem;`}
  ${props => props.py && `padding: ${props.py}rem 0rem;`}
  ${props => props.pt && `padding-top: ${props.pt}rem;`}
  ${props => props.pb && `padding-bottom: ${props.pb}rem;`}
  ${props => props.pr && `padding-right: ${props.pr}rem;`}
  ${props => props.pl && `padding-left: ${props.pl}rem;`}
  // background properties
  ${props => props.bg && `
    background-image: url(${props.src});
    background-repeat: no-repeat;
    background-attachment: fixed;
    background-size: cover;
    display: grid;
  `}
`;

export default Division;
