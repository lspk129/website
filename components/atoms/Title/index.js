import React from 'react';
import styled from 'styled-components';

const Title = styled(({ tag, children, ...props }) => React.createElement(tag, props, children))`
  text-align: center;

  ${({ tag, hero, theme }) => tag === 'h1' && `
    ${hero && `
      color: ${theme.text};
      font-size: 4rem;
      white-space: nowrap;
      font-weight: bold;
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.4);
      line-height: 4rem;
    `}
  `}

  ${({ tag, hero, theme }) => tag === 'h2' && `
    ${hero && `
      color: ${theme.text};
      text-shadow: 0 0.1rem 0.1rem rgba(0, 0, 0, 0.2);
    `}
  `}

  ${({ tag, upperCase }) => tag === 'h3' && `
    ${upperCase && 'text-transform: uppercase'}
  `}
`;

Title.defaultProps = {
  tag: 'h1',
};

export default Title;
