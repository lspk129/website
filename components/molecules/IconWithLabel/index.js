import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'react-styled-flexboxgrid';

import PaperIcons from '../../atoms/PaperIcons';
import Division from '../../atoms/Division';
import Title from '../../atoms/Title';

const IconWithLabel = ({ icon, label }) => (
  <Division m={2}>
    <Row center={'xs'}>
      <Col xs>
        <Row>
          <PaperIcons circle>
            {icon}
          </PaperIcons>
          <Division px={1}>
            <Title tag={'p'}>
              {label}
            </Title>
          </Division>
        </Row>
      </Col>
    </Row>
  </Division>
);

IconWithLabel.propTypes = {
  icon: PropTypes.node,
  label: PropTypes.string,
};

IconWithLabel.defaultProps = {
  icon: '',
  label: '',
};

export default IconWithLabel;
