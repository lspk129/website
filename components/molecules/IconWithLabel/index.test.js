import React from 'react';
import PropTypes from 'prop-types';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import injectTapEventPlugin from 'react-tap-event-plugin';
import { FaEnvelope } from 'react-icons/lib/fa';

import IconWithLabel from './';
import PaperIcons from '../../atoms/PaperIcons';
import Division from '../../atoms/Division';
import Title from '../../atoms/Title';

if (!process.tapEventInjected) {
  injectTapEventPlugin();
  process.tapEventInjected = true;
}

describe('IconWithLabel', () => {
  const muiTheme = getMuiTheme();
  const mountWithContext = node => mount(node, {
    context: { muiTheme },
    childContextTypes: { muiTheme: PropTypes.object },
  });
  const wrapper = mountWithContext(<IconWithLabel />);

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders with icon pops', () => {
    const wrapperWithProps = mountWithContext(
      <IconWithLabel icon={<FaEnvelope />} />,
    );
    expect(toJson(wrapperWithProps)).toMatchSnapshot();
  });

  it('renders with label props', () => {
    const wrapperWithProps = mountWithContext(
      <IconWithLabel label={'info@mokahaiku.com'} />,
    );
    expect(toJson(wrapperWithProps)).toMatchSnapshot();
  });

  it('renders IconWithLabel component', () => {
    expect(wrapper).toBePresent();
  });

  it('renders PaperIcons component', () => {
    expect(wrapper.find(PaperIcons)).toBePresent();
  });

  it('renders Division component', () => {
    expect(wrapper.find(Division)).toBePresent();
  });

  it('renders Paragraph component', () => {
    expect(wrapper.find(Title)).toBePresent();
  });
});
