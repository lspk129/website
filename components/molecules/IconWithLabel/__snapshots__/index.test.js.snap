// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconWithLabel renders correctly 1`] = `
<IconWithLabel
  icon=""
  label=""
>
  <styled.div
    m={2}
  >
    <div
      className="sc-bwzfXH eYmfyk"
    >
      <Row
        center="xs"
      >
        <div
          className="Row__Row-kNtYPA jaQNpG"
        >
          <Col
            xs={true}
          >
            <div
              className="Col__Col-hCQanm fvDNJi"
            >
              <Row>
                <div
                  className="Row__Row-kNtYPA bvOcuL"
                >
                  <Styled(Paper)
                    circle={true}
                  >
                    <Paper
                      circle={true}
                      className="sc-bdVaJa fGDlZB"
                      rounded={true}
                      transitionEnabled={true}
                      zDepth={1}
                    >
                      <div
                        className="sc-bdVaJa fGDlZB"
                        style={
                          Object {
                            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                            "backgroundColor": "#ffffff",
                            "borderRadius": "50%",
                            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                   0 1px 4px rgba(0, 0, 0, 0.12)",
                            "boxSizing": "border-box",
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontFamily": "Roboto, sans-serif",
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          }
                        }
                      />
                    </Paper>
                  </Styled(Paper)>
                  <styled.div
                    px={1}
                  >
                    <div
                      className="sc-bwzfXH cHqcyU"
                    >
                      <Styled(Component)
                        tag="p"
                      >
                        <Component
                          className="sc-htpNat iEBsLl"
                          tag="p"
                        >
                          <p
                            className="sc-htpNat iEBsLl"
                          />
                        </Component>
                      </Styled(Component)>
                    </div>
                  </styled.div>
                </div>
              </Row>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</IconWithLabel>
`;

exports[`IconWithLabel renders with icon pops 1`] = `
<IconWithLabel
  icon={<FaEnvelope />}
  label=""
>
  <styled.div
    m={2}
  >
    <div
      className="sc-bwzfXH eYmfyk"
    >
      <Row
        center="xs"
      >
        <div
          className="Row__Row-kNtYPA jaQNpG"
        >
          <Col
            xs={true}
          >
            <div
              className="Col__Col-hCQanm fvDNJi"
            >
              <Row>
                <div
                  className="Row__Row-kNtYPA bvOcuL"
                >
                  <Styled(Paper)
                    circle={true}
                  >
                    <Paper
                      circle={true}
                      className="sc-bdVaJa fGDlZB"
                      rounded={true}
                      transitionEnabled={true}
                      zDepth={1}
                    >
                      <div
                        className="sc-bdVaJa fGDlZB"
                        style={
                          Object {
                            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                            "backgroundColor": "#ffffff",
                            "borderRadius": "50%",
                            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                   0 1px 4px rgba(0, 0, 0, 0.12)",
                            "boxSizing": "border-box",
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontFamily": "Roboto, sans-serif",
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          }
                        }
                      >
                        <FaEnvelope>
                          <IconBase
                            viewBox="0 0 40 40"
                          >
                            <svg
                              fill="currentColor"
                              height="1em"
                              preserveAspectRatio="xMidYMid meet"
                              style={
                                Object {
                                  "color": undefined,
                                  "verticalAlign": "middle",
                                }
                              }
                              viewBox="0 0 40 40"
                              width="1em"
                            >
                              <g>
                                <path
                                  d="m40 15.8v17.8q0 1.4-1 2.5t-2.6 1h-32.8q-1.5 0-2.6-1t-1-2.5v-17.8q1 1.1 2.3 2 8 5.5 11 7.7 1.3 0.9 2.1 1.5t2.1 1 2.5 0.6h0q1.2 0 2.5-0.6t2.1-1 2.1-1.5q3.7-2.8 11.1-7.7 1.2-0.9 2.2-1.9z m0-6.5q0 1.7-1.1 3.4t-2.7 2.7q-8.4 5.8-10.5 7.3-0.2 0.1-0.9 0.6t-1.2 0.9-1.2 0.7-1.3 0.6-1.1 0.2h0q-0.5 0-1.1-0.2t-1.3-0.6-1.2-0.7-1.2-0.9-0.9-0.6q-2.1-1.5-5.9-4.1t-4.6-3.2q-1.3-0.9-2.6-2.6t-1.2-3q0-1.8 0.9-2.9t2.7-1.2h32.8q1.5 0 2.5 1.1t1.1 2.5z"
                                />
                              </g>
                            </svg>
                          </IconBase>
                        </FaEnvelope>
                      </div>
                    </Paper>
                  </Styled(Paper)>
                  <styled.div
                    px={1}
                  >
                    <div
                      className="sc-bwzfXH cHqcyU"
                    >
                      <Styled(Component)
                        tag="p"
                      >
                        <Component
                          className="sc-htpNat iEBsLl"
                          tag="p"
                        >
                          <p
                            className="sc-htpNat iEBsLl"
                          />
                        </Component>
                      </Styled(Component)>
                    </div>
                  </styled.div>
                </div>
              </Row>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</IconWithLabel>
`;

exports[`IconWithLabel renders with label props 1`] = `
<IconWithLabel
  icon=""
  label="info@mokahaiku.com"
>
  <styled.div
    m={2}
  >
    <div
      className="sc-bwzfXH eYmfyk"
    >
      <Row
        center="xs"
      >
        <div
          className="Row__Row-kNtYPA jaQNpG"
        >
          <Col
            xs={true}
          >
            <div
              className="Col__Col-hCQanm fvDNJi"
            >
              <Row>
                <div
                  className="Row__Row-kNtYPA bvOcuL"
                >
                  <Styled(Paper)
                    circle={true}
                  >
                    <Paper
                      circle={true}
                      className="sc-bdVaJa fGDlZB"
                      rounded={true}
                      transitionEnabled={true}
                      zDepth={1}
                    >
                      <div
                        className="sc-bdVaJa fGDlZB"
                        style={
                          Object {
                            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                            "backgroundColor": "#ffffff",
                            "borderRadius": "50%",
                            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
                                   0 1px 4px rgba(0, 0, 0, 0.12)",
                            "boxSizing": "border-box",
                            "color": "rgba(0, 0, 0, 0.87)",
                            "fontFamily": "Roboto, sans-serif",
                            "muiPrepared": true,
                            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                          }
                        }
                      />
                    </Paper>
                  </Styled(Paper)>
                  <styled.div
                    px={1}
                  >
                    <div
                      className="sc-bwzfXH cHqcyU"
                    >
                      <Styled(Component)
                        tag="p"
                      >
                        <Component
                          className="sc-htpNat iEBsLl"
                          tag="p"
                        >
                          <p
                            className="sc-htpNat iEBsLl"
                          >
                            info@mokahaiku.com
                          </p>
                        </Component>
                      </Styled(Component)>
                    </div>
                  </styled.div>
                </div>
              </Row>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </styled.div>
</IconWithLabel>
`;
